File: admin_web/lib/format.ts
const rupiahFormatter = new Intl.NumberFormat("id-ID", {
  style: "currency",
  currency: "IDR",
  minimumFractionDigits: 0,
  maximumFractionDigits: 0,
});

export function formatRupiah(value: number): string {
  return rupiahFormatter.format(value);
}

export function formatRupiahCompact(value: number): string {
  if (value >= 1_000_000_000) {
    return `Rp ${(value / 1_000_000_000).toFixed(1).replace(/\.0$/, "")}M`;
  }
  if (value >= 1_000_000) {
    return `Rp ${(value / 1_000_000).toFixed(1).replace(/\.0$/, "")}jt`;
  }
  if (value >= 1_000) {
    return `Rp ${(value / 1_000).toFixed(0)}rb`;
  }
  return `Rp ${value}`;
}

File: tamu_mobile/lib/shared/utils/format.dart
import 'package:intl/intl.dart';

final NumberFormat _rupiahFormatter = NumberFormat.currency(
  locale: 'id_ID',
  symbol: 'Rp ',
  decimalDigits: 0,
);

final DateFormat _tanggalFormatter = DateFormat('d MMMM yyyy', 'id_ID');
final DateFormat _tanggalPendekFormatter = DateFormat('d MMM yyyy', 'id_ID');

String formatRupiah(num value) => _rupiahFormatter.format(value);

String formatTanggal(DateTime tanggal) => _tanggalFormatter.format(tanggal);

String formatTanggalPendek(DateTime tanggal) =>
    _tanggalPendekFormatter.format(tanggal);
